source/daplink/interface/target_flash.c

```diff
@@ -253,16 +300,41 @@ static error_t target_flash_erase_chip(void)
 {
     if (g_board_info.target_cfg){
         error_t status = ERROR_SUCCESS;
-        const program_target_t *const flash = g_board_info.target_cfg->flash_algo;
-
-        status = flash_func_start(FLASH_FUNC_ERASE);
-        if (status != ERROR_SUCCESS) {
-            return status;
-        }
-        if (0 == swd_flash_syscall_exec(&flash->sys_call_s, flash->erase_chip, 0, 0, 0, 0)) {
-            return ERROR_ERASE_ALL;
+        program_target_t * flash, * start_flash_algo = last_flash_algo;
+        uint8_t index = 0;
+        
+        do{ 
+            flash = last_flash_algo;
+            status = flash_func_start(FLASH_FUNC_ERASE);
```

Comment:
Haha, you ran into the same thing I did in pyOCD! 😄 Do we assume each flash algo only erases that one region, or performs a full chip erase? There are almost certainly differences in how algos behave here. Eventually we need to add a flag that says whether the algo's erase all will erase the whole chip. But this is good for now, and is exactly what I did for pyOCD.